### DISCUSSION TITLE
Running DBADashService.exe in a docker container

### DISCUSSION THREAD

**Discussion by sporri** [trimble-oss/dba-dash#699] [Ideas]
I really would like to be able to run the DBADashService.exe in a docker container. The application is created as a windows service and will not work directly even though it can run on the command line on windows.

**Comment by DavidWiseman** [comment]
I haven't ran it on a docker container but it's possible to automate everything from the command line and I think it should work. You will need .NET 6 installed.  You run `DBADashService.exe` with the install command to install as a service [See here](https://docs.topshelf-project.com/en/latest/overview/commandline.html).  Also [here](https://dbadash.com/docs/setup/cli-install/) for additional commandline help.

Note: You will need to start the service after installing it.

If it's not working check the **Logs** folder. 

The "Configure & Install DBA Dash as Service" section of [this workflow](https://github.com/trimble-oss/dba-dash/blob/main/.github/workflows/ci.yml) also provides and example of some commandline automation.

Hope this helps.
